python/mxnet/ndarray/ndarray.py

```diff
@@ -102,6 +103,11 @@
 _NDARRAY_BASIC_INDEXING = 0
 _NDARRAY_ADVANCED_INDEXING = 1
 
+if sys.version_info[0] == 3:
+    _index_types = (int, np.int32, np.int64)
```

Comment:
why add a new one?

Reply:
sparse is using integer_types and I was afraid this change would break the use case there. Anyway, I finally removed this `_index_types` and added `np.int32` and `np.int64` directly to `integer_types`. Unit test for sparse use case is also added.